components/camel-qdrant/src/main/java/org/apache/camel/component/qdrant/QdrantProducer.java

```diff
@@ -235,6 +241,26 @@ private boolean delete(Exchange exchange, AsyncCallback callback) {
         return false;
     }
 
+    private boolean createCollection(Exchange exchange, AsyncCallback callback) {
+        final Message in = exchange.getMessage();
+        final VectorParams body = in.getBody(VectorParams.class);
+
+        ObjectHelper.notNull(body, "A payload of type VectorParams.class is expected");
```

Comment:
I tend to use the `QuadrantActionException` to include also the action that was caused the exception in the message

Reply:
In general, we have been using the `ObjectHelper.notNull` to handle cases like this. It currently does not support an specific exception like that, but I think I can make a modification to it if we want to use a custom exception.

Reply:
I'll propose a PR to core so we can use it like this:

```
ObjectHelper.notNull(body, "A payload of type VectorParams.class is expected", () -> new QdrantActionException(
                    QdrantAction.DELETE,
                    "A payload of type PointsSelector, PointId or Filter is expected"));
```

Reply:
fone with me, I tend to use some better/custom exception when a failure happen during routing as it is easier to spot the reason/context as opposed of a generic one

Reply:
Maybe we should try to avoid duplicating the message ? so something like  

```java 
ObjectHelper.notNull(
    body,
     () -> new QdrantActionException(
    	QdrantAction.DELETE,
        "A payload of type PointsSelector, PointId or Filter is expected"));
```

but I don't know

Reply:
I have always thought that an `exchange.setException(...)` was the idiomatic way of rising exceptions the camel way in async exchange processing... Most probably at the end of the day both ways result in the same behavior but still I find `exchange.setException(...)` more "camel idiomatic". 

My 2c.

Reply:
Actually the behavior is not the same. Using just `exchange.setException` (even followed by a return true/false) causes the code to crash/hang upon receiving a null body (it seems not to force the exchange to stop processing).

Recently, we have consolidated in using the `ObjectHelper.notNull` as the way to check for required parameters/body as it keeps the behavior and code base consistent.

Reply:
this is a mistake as the callback is not invoked when setting an exception which I'm fixing now

Reply:
The root of the problem might be that we are checking for requirements a bit too late (IMHO, we should check this before `process`) and we should leave the `setException` for processing exceptions.

Reply:
@lburgazzoli also, let's split the parameter checking from the processing exceptions, so it's consistent with the rest of the components. After you implement this fix, I'll adjust the code to move the parameter checking.

Reply:
I'm not sure I follow how that should be spitted to be honest

Reply:
> I'm not sure I follow how that should be spitted to be honest

I meant that we should be checking for the body, headers, etc before we actually get to do the processing (`createCollection`, `retrieve`, etc). But that's not necessary after you've changed the code to use the `try/catch` as @davsclaus suggested.